gobblin-runtime/src/main/java/gobblin/runtime/local/LocalJobManager.java

```diff
@@ -405,9 +405,9 @@ public synchronized void onTaskCompletion(String jobId, TaskState taskState) {
       return;
     }
 
-    if (JobMetrics.isEnabled(this.properties)) {
+    if (GobblinMetrics.isEnabled(this.properties)) {
       // Remove all task-level metrics after the task is done
-      taskState.removeMetrics();
```

Comment:
There is no implementation for removing metrics atm. When a metric context stops being used, it should be GCed automatically. I will remove the block.